services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -3720,14 +3720,19 @@ Pair<Store, Version> waitVersion(String clusterName, String storeName, int versi
   @Override
   public void setStoreCurrentVersion(String clusterName, String storeName, int versionNumber) {
```

Comment:
>
 Can you explain why non-zero parent current-version can affect other admin operations? Maybe we should enhance that piece of logic as well?

This is the code in `VeniceHelixAdmin#setStoreCurrentVersion`:
```
      int previousVersion = store.getCurrentVersion();
      store.setCurrentVersion(versionNumber);
      realTimeTopicSwitcher.transmitVersionSwapMessage(store, previousVersion, versionNumber);
```

Here, if current version in parent gets updated, and then the version cycles out of the parent Zk (through enough VPJ afterwards), then it causes exceptions in `RealTimeTopicSwitcher#transmitVersionSwap`:
```
    Version previousStoreVersion = store.getVersion(previousVersion)
        .orElseThrow(
            () -> new VeniceException(
                "Corresponding version " + previousVersion + " does not exist for store: " + store.getName()));
    Version nextStoreVersion = store.getVersion(nextVersion)
        .orElseThrow(
            () -> new VeniceException(
                "Corresponding version " + nextVersion + " does not exist for store: " + store.getName()));
```

This `RealTimeTopicSwitcher#transmitVersionSwap` is not supposed to execute in the parent controller. The short-circuit added in `VeniceHelixAdmin#setStoreCurrentVersion` will prevent this function from executing.

Reply:
I see, thanks for the explanation!